### PR TITLE
Make displayText in createNetwork optional

### PR DESCRIPTION
**Pull request by rahulbcn27**
### Description
This is a improvement change to make displayText field optional for createNetwork as mentioned in issue #6855. 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by rohityadavcloud** [comment]
@blueorangutan ui

**Comment by rohityadavcloud** [comment]
@blueorangutan ui

File: ui/src/views/network/CreateIsolatedNetworkForm.vue [hunk 1]
```diff
@@ -40,7 +40,7 @@
               <tooltip-label :title="$t('label.displaytext')" :tooltip="apiParams.displaytext.description"/>
             </template>
             <a-input
-             v-model:value="form.displaytext"
+             v-model:value="form.name"
```
Comment (DaanHoogland) [inline ui/src/views/network/CreateIsolatedNetworkForm.vue:43]: @rahulbcn27 , I don´t mean to take the name unconditionally, but only when the users decides to submit without a value added in the `displaytext` field.

I would recommend to make the field optional on the command and implement the check on the service.

**Comment by rohityadavcloud** [comment]
@blueorangutan ui

File: ui/src/views/network/CreateIsolatedNetworkForm.vue (hunk 1 above)
Comment (rahulbcn27) [reply ui/src/views/network/CreateIsolatedNetworkForm.vue:43]: Thanks @DaanHoogland for the comment. I am implementing the change suggested by you and Rohit.

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java [hunk 2]
```diff
@@ -218,7 +218,10 @@ public String getNetworkName() {
     }
 
     public String getDisplayText() {
-        return displayText;
+        if(displayText == null)
+            return name;
+        else
+            return displayText;
```
Comment (DaanHoogland) [inline api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:224]:
```suggestion
        return StringUtils.isEmpty(displayText) ? name : displayText;
```

**Review by DaanHoogland — COMMENTED** [review]
looks mostly good, I am not sure if my suggestion is needed. If empty string is checked for elsewhere, it isn´t. Please check or change.

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java (hunk 2 above)
Comment (rahulbcn27) [reply api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java]: Made the changes  as suggested.

**Comment by rahulbcn27** [comment]
@blueorangutan ui

**Comment by rahulbcn27** [comment]
@blueorangutan ui

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java (hunk 2 above)
Comment (DaanHoogland) [reply api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:225]: I am unresolving this @rahulbcn27 . I don't see the `==` replaced by `StringUtils.isEmpty(..)`

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java [hunk 3]
```diff
@@ -47,6 +47,7 @@
 import com.cloud.offering.NetworkOffering;
 import com.cloud.utils.net.NetUtils;
 
+
```
Comment (rohityadavcloud) [inline api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:50]: unnecessary change?

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java [hunk 4]
```diff
@@ -218,7 +219,10 @@ public String getNetworkName() {
     }
 
     public String getDisplayText() {
-        return displayText;
+        if(displayText == null)
```
Comment (rohityadavcloud) [inline api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:222]: nit - space between `if (`

**Review by rohityadavcloud — CHANGES_REQUESTED** [review]
I don't see UI changes.
- You can add unit test
- You need to ask somebody to manually QA your feature for UI changes which arent' covered by automated tests

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java (hunk 3 above)
Comment (rahulbcn27) [reply api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:50]: Fixed

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java (hunk 2 above)
Comment (rahulbcn27) [reply api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java]: Fixed as suggested

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java (hunk 4 above)
Comment (rahulbcn27) [reply api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java]: StringUtils.isEmpty(displayText) ? name : displayText; added as suggested by Daan.

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java (hunk 3 above)
Comment (rahulbcn27) [reply api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:50]: Fixed.

**Review by DaanHoogland — APPROVED** [review]
looks good, now we have to decide if we merge like this or apply this concequently throughout as requested in #6855 . I am fine with this, but it not closing [that](#6855)

**Comment by vladimirpetrov** [comment]
@blueorangutan package

**Comment by vladimirpetrov** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java [hunk 5]
```diff
@@ -61,7 +62,7 @@ public class CreateNetworkCmd extends BaseCmd implements UserCmd {
     @Parameter(name = ApiConstants.NAME, type = CommandType.STRING, required = true, description = "the name of the network")
     private String name;
 
-    @Parameter(name = ApiConstants.DISPLAY_TEXT, type = CommandType.STRING, required = true, description = "the display text of the network")
+    @Parameter(name = ApiConstants.DISPLAY_TEXT, type = CommandType.STRING, required = false, description = "the display text of the network")
```
Comment (stephankruggg) [inline api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:65]:
```suggestion
    @Parameter(name = ApiConstants.DISPLAY_TEXT, type = CommandType.STRING, description = "the display text of the network")
```

**Comment by vladimirpetrov** [comment]
Only 'Create network' has been fixed, here are the operations that still require 'displaytext' as a mandatory parameter:

- create disk offering
- create service (system, compute) offering
- create network offering
- create vpcoffering
- create project
- create vpc
- create template from volume
- register/upload template
- register/upload ISO

**Comment by DaanHoogland** [comment]
> @vladimirpetrov the issue is very general in nature; in the #6855 issue, Daan has only mentioned for the create Network form. I think we can start with that and explore all the other forms/actions you've identified.
> 
> @rahulbcn27 can you review the outstanding remarks and;
> 
>     * Update the PR subject to reflect what change you're doing
> 
>     * Open a new issue to log requests from Vladi (to address them) or ask @DaanHoogland  to update [UI: make display elements optional #6855](https://github.com/apache/cloudstack/issues/6855) the list
> 
>     * Add unit test or marvin test on your changes
> 
> 
> @blueorangutan package

@rahulbcn27 @rohityadavcloud I'll update the issue, create network was only meant to be an example. I revise the text if needed.

**Review by stephankruggg — APPROVED** [review]
CLGTM

**Comment by rahulbcn27** [comment]
> 

Thanks @DaanHoogland for updating the PR subject line.

File: test/integration/smoke/test_create_network.py [hunk 6]
```diff
@@ -289,3 +289,86 @@ def test_02_create_network_with_same_name(self):
...
+        Configurations.update(self.apiclient,
+                              name="allow.duplicate.networkname",
+                              value="true"
+                              )
+
+        # Create network offering
+        self.network_offering = NetworkOffering.create(
+            self.apiclient,
+            self.testdata["network_offering_shared"]
+        )
```
Comment (DaanHoogland) [inline test/integration/smoke/test_create_network.py:309]:
```suggestion
        self.network_offering = NetworkOffering.create(
            self.apiclient,
            self.testdata["network_offering_shared"]
        )
        self.cleanup.append( self.network_offering)
```

File: test/integration/smoke/test_create_network.py [hunk 7]
```diff
@@ -289,3 +289,86 @@ def test_02_create_network_with_same_name(self):
...
+            domainid=self.account.domainid
+        )
+
+        self.assertNotEqual(
+            network.displayText,
+            self.testdata["shared_network_sg"]["name"],
+            msg="displayText does not match name"
+        )
+
+        self.cleanup.append(network2)
```
Comment (DaanHoogland) [inline test/integration/smoke/test_create_network.py:369]:
```suggestion
        self.cleanup.append(network2)

        self.assertNotEqual(
            network.displayText,
            self.testdata["shared_network_sg"]["name"],
            msg="displayText does not match name"
        )

```

File: api/src/test/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmdTest.java [hunk 8]
```diff
@@ -100,12 +100,20 @@ public void testGetNetworkName() {
...
         Assert.assertEquals(cmd.getDisplayText(), description);
     }
 
+    public void testGetDisplayTextWhenEmpty() {
+        String description = null;
+        String netName = "net-isolated";
+        ReflectionTestUtils.setField(cmd, "displayText", description);
+        ReflectionTestUtils.setField(cmd, "name", netName);
+        Assert.assertEquals(cmd.getDisplayText(), netName);
+    }
```
Comment (DaanHoogland) [inline api/src/test/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmdTest.java:115]: i think this is not the behaviour we want to test. Instead I think we need to
```suggestion
    public void testGetDisplayTextWhenEmpty() {
        String description = null;
        String netName = "net-isolated";
        ReflectionTestUtils.setField(cmd, "name", netName);
        Assert.assertEquals(cmd.getDisplayText(), netName);
    }
```

**Comment by rohityadavcloud** [comment]
@DaanHoogland did we run smoketests on this?

**Comment by DaanHoogland** [comment]
oh, good point, we did but I don´t see results. i did run the unit test, but I'll restart teh smoke tests on the healcheck PR
